src/openrct2/drawing/Drawing.String.cpp

```diff
@@ -48,8 +48,7 @@ static int32_t TTFGetStringWidth(std::string_view text, FontStyle fontStyle, boo
  */
 int32_t GfxGetStringWidthNewLined(std::string_view text, FontStyle fontStyle)
 {
-    thread_local std::string buffer;
```

Comment:
I believe this was done to avoid allocations as it acts only as a temporary buffer. But I think I can let this slip until it shows up during a random profiling session again.

Reply:
Since this string gets resized to 0 anyways I'm not that convinced it saves any allocation time - as that 0 resize may be freeing the memory.

Reply:
resize/clear actually keeps the capacity, it won't destroy the buffer, you have to use https://en.cppreference.com/w/cpp/string/basic_string/shrink_to_fit in order to free unused capacity.

Reply:
I see, regardless if this is an issue in the future it may be a good idea to roll out a frame allocator/stack based string solution for such temporaries. A `std::string`-like type with a 256-512b big small string optimization space on the stack should cover 99.9% of the hot code paths. It's done this way e.g. in DuckStation, with success.

Reply:
I've been toying with the idea of having scratch buffers that are cleared after every frame is fully processed, however I would like to first have more code refurbished before adding more logic. At the moment this is not a noticeable issue so its not exactly high priority.